docs/book/src/developers/releasing.md

```diff
@@ -147,14 +147,17 @@ Here is a reference PR that applied the required test job changes following the
 
 ### Update Netlify branch (skip for patch releases)
 
-Go to [the Netlify branches and deploy contexts in site settings](https://app.netlify.com/sites/kubernetes-sigs-cluster-api-provider-azure/settings/deploys#branches-and-deploy-contexts) and click "edit settings". Update the "Production branch" to the new release branch and click "Save". The, go to the [Netlify site deploys](https://app.netlify.com/sites/kubernetes-sigs-cluster-api-provider-azure/deploys) and trigger a new deploy.
+Go to [the Netlify branches and deploy contexts in site settings](https://app.netlify.com/sites/kubernetes-sigs-cluster-api-provider-azure/settings/deploys#branches-and-deploy-contexts) and click "edit settings". Update the "Production branch" to the new release branch and click "Save". Then, go to the [Netlify site deploys](https://app.netlify.com/sites/kubernetes-sigs-cluster-api-provider-azure/deploys) and trigger a new deploy: Click on the "Trigger deploy" dropdown list and select "Deploy site".
 
 ![Netlify settings screenshot](images/netlify_deploys.png)
 
 Note: this step requires access to the Netlify site. If you don't have access, please ask a maintainer to update the branch.
 
 ### Communication
 
+1. Announce the release in Kubernetes Slack on the #cluster-api-azure channel.
+2. Send an announcement email to `kubernetes-sig-azure@googlegroups.com` and `kubernetes-sig-cluster-lifecycle@googlegroups.com` with the subject `[ANNOUNCE] cluster-api-provider-azure <version> has been released`
```

Comment:
Do we really want to send an email to the mailing list for patch releases?

Reply:
maybe this should be more inline with our current processes, i.e. announce the release at the next CAPZ office hours and add it to provider updates in CAPI office hours?

Reply:
I think we intended/should send an email only for minor releases, but announce all releases in Slack.

Reply:
Revisiting this after a while, thanks for the feedback!

I'm not clear why we would have a preferred/expanded announcement process for minor over patch releases. Seems like "I've been waiting on feature X which is in the next minor release" and "I've been waiting for a fix to Y" which will be in the next patch release" deserve equivalent broadcasting, IMO.

Reply:
I think it might be just because patch releases are way more often so we don't want to cause too much noise on the mailing list?

Reply:
Cool, that makes sense. My one vote goes towards not worrying about noise and preferring consistency, but let's see if I'm outvoted!

Reply:
I agree with Cecile: I think the quieter notifications for patch releases were intended to reduce noise on the mailing list. Having said that, I don't feel strongly either way and simplification / consistency is a good argument.

Reply:
I think we don't have sufficient consensus to make any changes at this time. I'll close this PR!